src/postamble.js

```diff
@@ -66,20 +65,37 @@ if (memoryInitializer) {
     }
     function doBrowserLoad() {
       Module['readAsync'](memoryInitializer, applyMemoryInitializer, function() {
+#if SUPPORT_BASE64_EMBEDDING
+        var memoryInitializerBytes = tryParseAsDataURI(memoryInitializer);
+        if (memoryInitializerBytes) {
+          applyMemoryInitializer(memoryInitializerBytes.buffer);
+          return;
+        }
+#endif
         throw 'could not load memory initializer ' + memoryInitializer;
       });
     }
     if (Module['memoryInitializerRequest']) {
       // a network request has already been created, just use that
       function useRequest() {
         var request = Module['memoryInitializerRequest'];
+        var response = request.response;
         if (request.status !== 200 && request.status !== 0) {
-          // If you see this warning, the issue may be that you are using locateFile or memoryInitializerPrefixURL, and defining them in JS. That
-          // means that the HTML file doesn't know about them, and when it tries to create the mem init request early, does it to the wrong place.
-          // Look in your browser's devtools network console to see what's going on.
-          console.warn('a problem seems to have happened with Module.memoryInitializerRequest, status: ' + request.status + ', retrying ' + memoryInitializer);
-          doBrowserLoad();
-          return;
+#if SUPPORT_BASE64_EMBEDDING
+          var data = tryParseAsDataURI(Module['memoryInitializerRequestURL']);
+          if (data) {
+            response = data.buffer;
+          } else {
+#endif
+            // If you see this warning, the issue may be that you are using locateFile or memoryInitializerPrefixURL, and defining them in JS. That
+            // means that the HTML file doesn't know about them, and when it tries to create the mem init request early, does it to the wrong place.
+            // Look in your browser's devtools network console to see what's going on.
+            console.warn('a problem seems to have happened with Module.memoryInitializerRequest, status: ' + request.status + ', retrying ' + memoryInitializer);
+            doBrowserLoad();
+            return;
+#if SUPPORT_BASE64_EMBEDDING
+          }
+#endif
         }
         applyMemoryInitializer(request.response);
```

Comment:
shouldn't this line use `response`? not just because it's shorter, but the code above assigns to it, I assume with the intention that it be used here.

Reply:
Ah yeah, that does look more correct. I must've just missed that when adding the base64 stuff. Fixed.